feat(accordion): added display-lg, support multiple bodies

mcoker opened the pull request:
fixes https://github.com/patternfly/patternfly/issues/3865

* Adds web/marketing variation
* Moves the pseudo elements around a little to support the border overlap
* Adds support for multiple `__expanded-content-body` elements

@ajacobs21e @mcarrano can you also verify the spacing between adjacent sections is correct in the regular/fixed example? https://patternfly-pr-3888.surge.sh/components/accordion#fixed

patternfly-build commented:
Preview: https://patternfly-pr-3888.surge.sh

mcarrano reviewed (commented):
Looks good to me @mcoker . I'll defer to @ajacobs21e regarding the spacing and font sizes for the new Large variant. Not sure how to answer your question regarding the "fixed" variant. Looks OK to me but perhaps @mceledonia should also take a look.

mceledonia commented:
Looks alright to me @mcarrano @ajacobs21e !

ajacobs21e commented:
LGTM! https://www.sketch.com/s/00432cc6-fd0c-454f-bdd6-c9720ceebbdc/a/OvQza4

mcoker commented:
> Not sure how to answer your question regarding the "fixed" variant.

Currently the accordion component only accepts a single body element per accordion item. So all of the content in the accordion item goes in a single accordion item body. This PR introduces the ability to pass multiple body elements to an accordion item, so just wanted to verify the 8px spacer between body items in the regular accordion (not the large variation introduced here) looks OK.

mcarrano commented:
Ah, I see @mcoker . I think that the 8px spacer looks fine in this example. So would the spacing always be 8px or could I define a larger spacing if I wanted. Is the 8px consistent with the spacing that we use on a standard form?

mcoker commented:
> would the spacing always be 8px or could I define a larger spacing if I wanted

The spacing between accordion body elements would always be 8px (or whatever space we define here). If you wanted different spacing, you could always use a stack or flex or other layout in a single accordion body and define your own spacing. 

> Is the 8px consistent with the spacing that we use on a standard form?

The spacer between form rows is 16px. I used an 8px spacer here since the spacer in the large/website variation is 16px, but we could use that as the non-large spacer, too.

current/8px spacer
<img width="848" alt="Screen Shot 2021-03-01 at 2 22 12 PM" src="https://user-images.githubusercontent.com/35148959/109554318-af072c80-7a99-11eb-88cd-36ca20b11a81.png">


16px spacer
<img width="863" alt="Screen Shot 2021-03-01 at 2 22 59 PM" src="https://user-images.githubusercontent.com/35148959/109554336-b29ab380-7a99-11eb-8a13-8012d15fc45d.png">

mattnolting reviewed (commented):
Looks great, can we separate `display-lg` and `pf-m-border`?

mcoker commented:
> Looks great, can we separate display-lg and pf-m-border?

@ajacobs21e @mcarrano wdyt? Should the larger presentation and the borders between the items be separate variations? That implies that you could use the bordered variation with a regular accordion, and could use the large/web variation without a border - is that what you were thinking @mattnolting?

mattnolting commented:
@mcoker yessir, that's exactly it!

mcarrano commented:
@mcoker @mattnolting that makes sense to me. @mceledonia can you see any reason we shouldn't allow borders on the standard/regular accordion headers?

mcoker commented:
@mcarrano @mattnolting separated bordered & large variations, added a regular bordered example and updated the large example to include the bordered variation.

mcarrano reviewed (approved):
Looks great @mcoker . I actually like this much better with the borders on both regular and large variants.

@doruskova let's make sure to add this option to the Sketch kit, also. Can you open an issue? Thanks.

mattnolting reviewed (approved):
Fanstatic! 🚢  it!

mcoker commented:
>  I actually like this much better with the borders on both regular and large variants.

@mattnolting++

redallen commented:
:tada: This PR is included in version 4.90.0 :tada:

The release is available on:
- [GitHub release](https://github.com/patternfly/patternfly/releases/tag/prerelease-v4.90.0)
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@patternfly/patternfly/v/4.90.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
